libvirt/tests/src/virsh_cmd/domain/virsh_define.py

```diff
@@ -106,6 +108,15 @@ def prepare_xml_file():
             virsh.define(xml_backup_file)
         return
 
+    if is_testing_timeout:
+        virsh.undefine(vm_name)
+        start_time = time.time()
+        virsh.define(xml_backup_file)
```

Comment:
We have 2 more options - set timeout for 'virsh define' or check the command execution duration. 
eg, set timeout 3 for virsh command  
```suggestion
        result = virsh.define(xml_backup_file, timeout=3)
        libvirt.check_result(result)
```

or
```suggestion
        result = virsh.define(xml_backup_file)
        if result.duration > 3:
           xxx
```

Reply:
Thanks! I have updated it with way 2. It's more simpler. 

btw, The way 1 doesn't seem to throw a timeout exception.The following is the job log.
```
2022-02-14 01:20:30,531 virsh            L1581 DEBUG| Undefine VM avocado-vt-vm1
2022-02-14 01:20:30,583 virsh            L1561 DEBUG| Define VM from /tmp/xml_utils_temp_nmtb18jo.xml
2022-02-14 01:20:33,587 process          L0604 INFO | Running 'kill -19 75569'
2022-02-14 01:20:33,591 process          L0677 INFO | Command kill -19 75569 running on a thread
2022-02-14 01:20:33,591 process          L0686 INFO | Command 'kill -19 75569' finished with 0 after 0.0009262561798095703s
2022-02-14 01:20:33,629 process          L0604 INFO | Running 'kill -15 75569'
2022-02-14 01:20:33,632 process          L0677 INFO | Command kill -15 75569 running on a thread
2022-02-14 01:20:33,632 process          L0686 INFO | Command 'kill -15 75569' finished with 0 after 0.0009253025054931641s
2022-02-14 01:20:33,633 process          L0604 INFO | Running 'kill -18 75569'
2022-02-14 01:20:33,635 process          L0677 INFO | Command kill -18 75569 running on a thread
2022-02-14 01:20:33,636 libvirt          L1248 DEBUG| Command result:

2022-02-14 01:20:33,636 process          L0686 INFO | Command 'kill -18 75569' finished with 0 after 0.001455068588256836s
2022-02-14 01:20:33,638 stacktrace       L0039 ERROR|
2022-02-14 01:20:33,638 stacktrace       L0042 ERROR| Reproduced traceback from: /root/avocado-build/avocado_vt/avocado_vt/test.py:275
2022-02-14 01:20:33,640 stacktrace       L0045 ERROR| Traceback (most recent call last):
2022-02-14 01:20:33,640 stacktrace       L0045 ERROR|   File "/var/lib/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/virsh_cmd/domain/virsh_define.py", line 113, in run
2022-02-14 01:20:33,640 stacktrace       L0045 ERROR|     libvirt.check_result(result)
2022-02-14 01:20:33,641 stacktrace       L0045 ERROR|   File "/root/avocado-build/avocado_vt/virttest/utils_test/libvirt.py", line 1284, in check_result
2022-02-14 01:20:33,641 stacktrace       L0045 ERROR|     "Expect should succeed, but got: %s" % all_msg)
2022-02-14 01:20:33,641 stacktrace       L0045 ERROR| avocado.core.exceptions.TestFail: Expect should succeed, but got:
2022-02-14 01:20:33,641 stacktrace       L0045 ERROR|
2022-02-14 01:20:33,641 stacktrace       L0046 ERROR|
2022-02-14 01:20:33,806 libvirt_vm       L2254 DEBUG| Releasing MAC addresses for vm avocado-vt-vm1.
```